app/vmui/packages/vmui/src/components/CustomPanel/Views/TracingsView.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {FC} from "preact/compat";
+import {TracingData} from "../../../api/types";
+import Typography from "@mui/material/Typography";
+import TraceView from "./TraceView";
+import Alert from "@mui/material/Alert";
+import RemoveCircleIcon from "@mui/icons-material/RemoveCircle";
+import Button from "@mui/material/Button";
+
+interface TraceViewProps {
+  tracingsData: TracingData[];
+  emptyMessage: string;
+  onDeleteClick: (tracingData: TracingData) => void;
+}
+
+const TracingsView: FC<TraceViewProps> = ({tracingsData, emptyMessage, onDeleteClick}) => {
+  if (!tracingsData.length) {
+    return (
+      <>
+        <Alert color={"info"} severity="info" sx={{whiteSpace: "pre-wrap", mt: 2}}>
+          {emptyMessage}
+        </Alert>
+      </>
+    );
+  }
+
+  const handleDeleteClick = (tracingData: TracingData) => () => {
+    onDeleteClick(tracingData);
+  };
+
+  const getQuery = (message: string): string => {
+    const query = message.match(/query=(.*):/);
```

Comment:
This looks really horrible :)

It would be better just storing and using the original `query` from a `Trace` object as described in previous comments.

Reply:
used call method from Trace model